tests/lib/rules/newline-before-return.js

```diff
@@ -112,77 +112,90 @@ ruleTester.run("newline-before-return", rule, {
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nfor (b in c) {\nd();\n\nreturn;\n}\n}"
         },
         {
             code: "function a() {\nfor (b of c) {\nd();\nreturn;\n}\n}",
             parserOptions: { ecmaVersion: 6 },
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nfor (b of c) {\nd();\n\nreturn;\n}\n}"
         },
         {
             code: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}",
-            errors: ["Expected newline before return statement."]
-        },
-        {
-            code: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) {\nc();\n}\n//comment\nreturn b;\n}"
         },
         {
             code: "function a() {\n/*comment\ncomment*/\nif (b) {\nc();\nreturn b;\n} else {\n//comment\n\nreturn d;\n}\n/*multi-line\ncomment*/\nreturn e;\n}",
-            errors: ["Expected newline before return statement.", "Expected newline before return statement."]
+            errors: ["Expected newline before return statement.", "Expected newline before return statement."],
+            output: "function a() {\n/*comment\ncomment*/\nif (b) {\nc();\n\nreturn b;\n} else {\n//comment\n\nreturn d;\n}\n/*multi-line\ncomment*/\nreturn e;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } //comment\nreturn c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } //comment\n\nreturn c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } /*multi-line\ncomment*/\nreturn c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } /*multi-line\ncomment*/\nreturn c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; }\n/*multi-line\ncomment*/ return c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; }\n/*multi-line\ncomment*/ return c;\n}"
         },
         {
             code: "function a() {\nif (b) { return; } /*multi-line\ncomment*/ return c;\n}",
-            errors: ["Expected newline before return statement."]
+            errors: ["Expected newline before return statement."],
+            output: "function a() {\nif (b) { return; } /*multi-line\ncomment*/ \nreturn c;\n}"
```

Comment:
Is there a reason we can't just check the `leadingComments` property of the node rather than the `calcCommentLines` function ? Would be more reliable and I think would fix this issue.